src/gui/qgsmapcanvas.cpp

```diff
@@ -778,8 +796,26 @@ QgsRectangle QgsMapCanvas::imageRect( const QImage& img, const QgsMapSettings& m
 
 void QgsMapCanvas::mapUpdateTimeout()
 {
-  const QImage& img = mJob->renderedImage();
-  mMap->setContent( img, imageRect( img, mJob->mapSettings() ) );
+  QImage img = mJob->renderedImage();
+  magnifyImage( img );
+
+  mMap->setContent( img, imageRect( img, mSettings ) );
+}
+
+void QgsMapCanvas::magnifyImage( QImage &img )
+{
+  if ( mMagnificationLevel != 1.0 )
+  {
+    img = img.scaled( img.width() * mMagnificationLevel,
```

Comment:
+1 to Nyalls comments

On Tue, Apr 19, 2016 at 7:22 PM, Alexandre Neto <notifications@github.com>
wrote:

> In src/gui/qgsmapcanvas.cpp
> <https://github.com/qgis/QGIS/pull/3009#discussion_r60198211>:
>
> > @@ -778,8 +796,26 @@ QgsRectangle QgsMapCanvas::imageRect( const QImage& img, const QgsMapSettings& m
> >
> >  void QgsMapCanvas::mapUpdateTimeout()
> >  {
> > -  const QImage& img = mJob->renderedImage();
> > -  mMap->setContent( img, imageRect( img, mJob->mapSettings() ) );
> > +  QImage img = mJob->renderedImage();
> > +  magnifyImage( img );
> > +
> > +  mMap->setContent( img, imageRect( img, mSettings ) );
> > +}
> > +
> > +void QgsMapCanvas::magnifyImage( QImage &img )
> > +{
> > +  if ( mMagnificationLevel != 1.0 )
> > +  {
> > +    img = img.scaled( img.width() * mMagnificationLevel,
>
>
>    - 1 Nyall
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/3009/files/e378ed5fff320f16970f638821ba4e5d51f6b0d8#r60198211>
>